### DISCUSSION TITLE
XMake 生成的Debug工程编译链接 使用tcmalloc反而比不使用更慢

### DISCUSSION THREAD

**Discussion by mengxinxiaoqiang** [xmake-io/xmake#4305] [Q&A] [answered]
同样的代码 
工程A：VS2022默认生成的控制台项目
工程B：XMake生成
![image](https://github.com/xmake-io/xmake/assets/56524631/d3429107-0497-473e-893a-88a41fa33406)
测试代码
![image](https://github.com/xmake-io/xmake/assets/56524631/385a64dc-72df-403a-961e-0e3ce52c5272)
A工程结果
开启__tcmalloc
![image](https://github.com/xmake-io/xmake/assets/56524631/52b09f89-402c-457a-b591-83f881513fd9)
关闭__tcmalloc
![image](https://github.com/xmake-io/xmake/assets/56524631/25d9965c-5560-45c9-909b-a153ee6ec1f8)
B工程结果
开启__tcmalloc
![image](https://github.com/xmake-io/xmake/assets/56524631/c48ea921-a14a-4224-b05e-c538ee42405e)
关闭__tcmalloc
![image](https://github.com/xmake-io/xmake/assets/56524631/6014a8b2-e257-49bf-8361-ba5ed17b0dc0)
均处于Windows Debug X64环境
请问各位大佬，是有什么flag不一样吗？还是其他问题，或者如何看到XMake 链接执行的cmd?

**Comment by star-hengxing** [comment]
`xmake -v`看编译命令

**Comment by mengxinxiaoqiang** [comment]
没看出有何不同的，两个工程都不使用tcmalloc，跑测试代码也有很大差距，希望有大佬能解惑

**Reply by star-hengxing** [reply, marked as answer]
如果确定编译命令一样，那应该和 xmake 无关。替换 malloc 库估计跟 crt 有关，而 vs 默认工程是 `MD`，xmake 默认是 `MT`
